src/features/allDataState/index.js

```diff
@@ -0,0 +1,297 @@
+import { createSelector } from 'reselect'
+
+const selectGroceriesStatus = (state) => state.groceries.dataStatus
+const selectProductsStatus = (state) => state.products.statusData
+const selectDishesStatus = (state) => state.dishes.dataStatus
+const selectSavedListStatus = (state) => state.savedList.statusData
+const selectParentCategoriesStatus = (state) => state.parentCategories.status
+const selectHistoryStatus = (state) => state.history.status
+const selectSettingsStatus = (state) => state.settings.status
+
+export const selectGroceriesBasketData = (state) => state.groceries.basketData
+export const selectProductsBasketData = (state) => state.products.basketData
+export const selectDishesBasketData = (state) => state.dishes.basketData
+export const selectSavedListBasketData = (state) => state.savedList.basketData
+
+export const isBasketEmpty = createSelector(
+  selectGroceriesBasketData,
+  selectProductsBasketData,
+  selectDishesBasketData,
+  selectSavedListBasketData,
+
+  (
+    selectGroceriesBasketData,
+    selectProductsBasketData,
+    selectDishesBasketData,
+    selectSavedListBasketData,
+  ) => {
+    const empty =
+      selectGroceriesBasketData.length === 0 &&
+      selectProductsBasketData.length === 0 &&
+      selectDishesBasketData.length === 0 &&
+      selectSavedListBasketData.length === 0
+
+    return empty
+  },
+)
+
+export const itemsDataStatus = createSelector(
+  selectGroceriesStatus,
+  selectProductsStatus,
+  selectDishesStatus,
+  selectSavedListStatus,
+  selectParentCategoriesStatus,
+  isBasketEmpty,
+
+  (
+    selectGroceriesStatus,
+    selectProductsStatus,
+    selectDishesStatus,
+    selectSavedListStatus,
+    selectParentCategoriesStatus,
+    isBasketEmpty,
+  ) => {
+    const dataStatus = [
+      selectGroceriesStatus,
+      selectProductsStatus,
+      selectDishesStatus,
+      selectSavedListStatus,
+      selectParentCategoriesStatus,
+    ]
+
+    let result = {
+      isError: false,
+      isLoading: true,
+      isLoaded: false,
+      isBasketEmpty: false,
+    }
+
+    let workingIsLoaded = true
+    result.isBasketEmpty = isBasketEmpty
+
+    dataStatus.forEach((singleStatus) => {
```

Comment:
tutaj mozna poprawic wydajnosc. za malo danych zeby to mialo znaczenie, ale ladniej byloby jakbys po kolei robił .some, np. 

if(dataStatus.some(status => status === 'failed') {
result.isError = true
}

ale tak jak nizej, lepiej to uproscic zupelnie

Reply:
Poprawione.